op-monitorism/global_events/monitor.go

```diff
@@ -0,0 +1,238 @@
+package global_events
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-service/metrics"
+	"github.com/ethereum/go-ethereum"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	git "github.com/go-git/go-git/v5"
+
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/prometheus/client_golang/prometheus"
+	"time"
+)
+
+const (
+	MetricsNamespace = "global_events_mon"
+)
+
+// Monitor is the main struct of the monitor.
+type Monitor struct {
+	log log.Logger
+
+	l1Client     *ethclient.Client
+	globalconfig GlobalConfiguration
+	// nickname is the nickname of the monitor (we need to change the name this is not an ideal one here).
+	nickname    string
+	safeAddress *bindings.OptimismPortalCaller
+
+	LiveAddress *common.Address
+
+	filename   string //filename of the yaml rules
+	yamlconfig Configuration
+
+	// Prometheus metrics
+	eventEmitted        *prometheus.GaugeVec
+	unexpectedRpcErrors *prometheus.CounterVec
+}
+
+// ChainIDToName() allows to convert the chainID to a human readable name.
+// For now only ethereum + Sepolia are supported.
+func ChainIDToName(chainID int64) string {
+	switch chainID {
+	case 1:
+		return "Ethereum [Mainnet]"
+	case 11155111:
+		return "Sepolia [Testnet]"
+	}
+	return "The `ChainID` is Not defined into the `chaindIDToName` function, this is probably a custom chain otherwise something is going wrong!"
+}
+
+func cloneRepo(repoURL string, path string) error { //for debug purpose will be store to /tmp/Monitorism/rules_l1
```

Comment:
I would remove this. It seems unnecessary at this stage and the function is not managing errors in which the clone may go wrong (for example when already exists, it fails but should probably not).
Also branch is hardcoded, it seems opinionated and not generic enough.

Reply:
I have commented this function.